### PR TITLE
Few changes to the configmap package

### PR DESCRIPTION
**Pull request by dprotaso**
- New* methods return concrete types
- defaultImpl is now InformedWatcher
- fixedImpl is now StaticWatcher
- Included a new ManualWatcher that mimics the InformedWatcher
  in behaviour. ie. allows updates - making this a true 'mock'

### CODE REVIEW

File: configmap/informed_watcher.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	}
+}
+
+// InformedWatcher provides an informer-based implementation of Watcher.
+type InformedWatcher struct {
+	sif      informers.SharedInformerFactory
+	informer corev1informers.ConfigMapInformer
+	started  bool
+
+	ManualWatcher
```
Comment (mattmoor) [inline configmap/informed_watcher.go]: What is this doing for us?

**Comment by dprotaso** [comment]
Serving - https://github.com/knative/serving/pull/2006
Eventing - https://github.com/knative/eventing/pull/428

**Pull request merged**